crates/plugins/src/version_check.rs

```diff
@@ -0,0 +1,67 @@
+use crate::{
+    get_manifest_file_name, plugin_manifest::PluginManifest, PLUGIN_MANIFESTS_DIRECTORY_NAME,
+};
+use anyhow::{anyhow, Result};
+use semver::{Version, VersionReq};
+use std::{fs::File, path::Path};
+
+/// Checks whether the plugin supports the currently running version of Spin.
+pub fn assert_supported_version(spin_version: &str, supported: &str) -> Result<()> {
+    let supported = VersionReq::parse(supported).map_err(|e| {
+        anyhow!(
+            "could not parse manifest compatibility version {} as valid semver -- {:?}",
+            supported,
+            e
+        )
+    })?;
```

Comment:
Is there a way for the user to go "okay there's a typo in the version spec but I want it anyway"?